src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/PoolConstant.java

```diff
@@ -221,4 +222,57 @@ public Object poolKey(Types types) {
             return new Pair<>(name, new UniqueType(type, types));
         }
     }
+
+    /**
+     * A pool constant implementation describing a parameter pool entry.
+     */
+    final class Parameter implements PoolConstant {
+
+        final int kind;
+
+        Parameter(int kind) {
+            this.kind = kind;
+        }
+
+        @Override
+        public int poolTag() {
+            return ClassFile.CONSTANT_Parameter;
+        }
+
+        @Override
+        public Object poolKey(Types types) {
+            return Integer.valueOf(kind);
```

Comment:
This is not good enough - as it means that you can have only one parameter of each kind.
See also my other related comment on this topic. I think this Parameter entry should have a "fake" String id, so that we can give them a name, and distinguish between them; then you can just use the String id as a pool key.